version.json

```diff
@@ -0,0 +1,3 @@
+{
+  "version": "0.1.1"
```

Comment:
That sounds like a very random version number to start with. Even if we all agree that bumping the major version from 0 to 1 should happen at release, what to you think is the criteria for bumping minor and patch?

I don't understand how all these github actions work or interact. Is there some way this code associates either an automatically incremented build number (like a traditional CI build system like Jenkins), or an arbitrarily/random (e.g. latest git hash) build number? Or is this the version string in it's entirety? Then it either needs to be updated for every push, or we will publish multiple builds with the same version string. Neither is ideal in any way.

Reply:
Version bumps are dictated by:
```
fix: fixed NPE in class (increases the REVISION number by 1)
feat: Added minimap (increases the MINOR number by 1)
chore: fixed typo (does not increase version number)
ci: added github actions (does not increase version number)
docs: updated readme (does not increase version number)
```

Reply:
So how does that work? Do the changelog action push commits changing version.json?

Also, you realize this will mean we will end up with version numbers like `0.29.301`? (Traditional semver assumed that the last part were updated for a new release, not for every committed fix)